src/main/java/org/kuali/kfs/coa/document/validation/impl/AccountGlobalRule.java

```diff
@@ -863,5 +962,1065 @@ private boolean isIncomeStreamAccountRequired(String fundGroupCode, String subFu
+        boolean success = true;
+        if (StringUtils.isBlank(subFundGroupCode)) {
+            if (StringUtils.isNotBlank(appropriationAccountNumber)) {                    
+                AppropriationAccount appropriationAcct = getMatchedRecord(AppropriationAccount.class, account.getSubFundGroupCode(), CUKFSPropertyConstants.APPROPRIATION_ACCT_NUMBER, appropriationAccountNumber);
+                if (appropriationAcct == null) {
+                    success = false;
+                    GlobalVariables.getMessageMap().putError(KFSPropertyConstants.ACCOUNT_NUMBER, CUKFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_ACCT_APPROP_ACCT_NOT_GROUP_CODE, 
+                                new String[] {appropriationAccountNumber, account.getSubFundGroupCode(), account.getAccountNumber()});
+                } else {
+                    if (!appropriationAcct.isActive()) {
+                        putFieldError(CUKFSPropertyConstants.APPROPRIATION_ACCT_NUMBER, KFSKeyConstants.ERROR_INACTIVE, 
+                                getFieldLabel(AccountGlobal.class, CUKFSPropertyConstants.APPROPRIATION_ACCT_NUMBER));
+                        success = false;
+                    }
+                }                
+            }            
+        } else {
+            AccountExtendedAttribute accountExtension = (AccountExtendedAttribute)account.getExtension(); 
+            if (StringUtils.isBlank(appropriationAccountNumber)) {
+                if (StringUtils.isNotBlank(accountExtension.getAppropriationAccountNumber())) {
+                    AppropriationAccount appropriationAcct = getMatchedRecord(AppropriationAccount.class, subFundGroupCode, CUKFSPropertyConstants.APPROPRIATION_ACCT_NUMBER, accountExtension.getAppropriationAccountNumber());
+                    if (appropriationAcct == null) {
+                        success = false;
+                        GlobalVariables.getMessageMap().putError(KFSPropertyConstants.ACCOUNT_NUMBER, CUKFSKeyConstants.ERROR_DOCUMENT_ACCMAINT_ACCT_APPROP_ACCT_NOT_GROUP_CODE, 
+                                    new String[] {accountExtension.getAppropriationAccountNumber(), subFundGroupCode, account.getAccountNumber()});
+                    }                           
+                }
+            }
+        }        
+        return success;
+    }
+    
+    private <T extends BusinessObject> T getMatchedRecord(Class<T> clazz, String subFundGroupCode, String propertyName, String propertyValue) {
+        Map<String, String> fieldValues = new HashMap<String, String>();
+        fieldValues.put(KFSPropertyConstants.SUB_FUND_GROUP_CODE, subFundGroupCode);
+        fieldValues.put(propertyName, propertyValue);
+        
+        return getBoService().findByPrimaryKey(clazz, fieldValues);
+
+    }
+    
+	protected boolean checkOpenEncumbrances() {
+		boolean success = true;
+		for (AccountGlobalDetail detail : newAccountGlobal.getAccountGlobalDetails()) {
+			success &= checkOpenEncumbrances(detail);
```

Comment:
Thank you for pointing this out. I will add a new error message that it includes the account that has open encumbrances. This way the user will know which accounts cannot be closed from the ones selected on the doc and can make a decision on how to proceed based on that.